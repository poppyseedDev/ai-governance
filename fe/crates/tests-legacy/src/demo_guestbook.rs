#![cfg(feature = "solc-backend")]

use fe_compiler_test_utils::*;
use insta::assert_snapshot;

#[test]
fn guest_book() {
    with_executor(&|mut executor| {
        let mut harness = deploy_contract(&mut executor, "demos/guest_book.fe", "GuestBook", &[]);

        let sender = address_token("1234000000000000000000000000000000005678");
        let msg = string_token("hello world");

        harness.caller = sender.clone().into_address().unwrap();

        harness.test_function(&mut executor, "sign", &[msg.clone()], None);

        harness.test_function(&mut executor, "get_msg", &[sender], Some(&msg));

        harness.events_emitted(executor, &[("Signed", &[msg])]);

        assert_harness_gas_report!(harness);
    })
}
