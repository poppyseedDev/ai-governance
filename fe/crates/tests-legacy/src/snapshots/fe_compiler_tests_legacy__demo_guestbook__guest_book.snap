---
source: crates/tests-legacy/src/demo_guestbook.rs
expression: "format!(\"{}\", harness.gas_reporter)"
---
sign([String("hello world")]) used 54893 gas
get_msg([Address(0x1234000000000000000000000000000000005678)]) used 1786 gas

